src/shogun/mathematics/linalg/backend/eigen/SpecialPurposes.cpp

```diff
@@ -157,6 +157,18 @@ void LinalgBackendEigen::rectified_linear_impl(
 		result_eig(i) = Math::max((T)0, a_eig(i));
 }
 
+template <typename T>
+void LinalgBackendEigen::mish_impl(
+    const SGMatrix<T>& a, SGMatrix<T>& result) const
+{
+	typename SGMatrix<T>::EigenMatrixXtMap a_eig = a;
+	typename SGMatrix<T>::EigenMatrixXtMap result_eig = result;
+
+	for (index_t i = 0; i < a_eig.rows() * a_eig.cols(); ++i)
+		result_eig(i) = a_eig(i) * std::tanh(std::log(1 + std::exp(a_eig(i))));
```

Comment:
can you not write this as something like `result_eig = a_eig.array() * ((1 + a_eig.exp()).log()).tanh()`? That way you give eigen the best chance to vectorise this.

Reply:
`log1p` instead of `log(1 + exp(x))` is better for numerical accuracy but is still overall poor.

OpenCV CPU backend has a much faster (more than 2x) and accurate [implementation](https://github.com/opencv/opencv/blob/4564b8a224545a54c545e9d2743cce3c0e1f1c5b/modules/dnn/src/layers/elementwise_layers.cpp#L760-L773).

Reply:
do you know what the accuracy tradeoff is?

Reply:
![errors](https://user-images.githubusercontent.com/6652578/87765352-500be480-c835-11ea-844a-f65f40106aa3.png)

YOLOv4 performance almost doubles: https://github.com/opencv/opencv/pull/17624

Reply:
we could have the approximation somehwere, but it should be marked as such
(and I agree with @gf712 this can be vectorized, even the approximation)